src/common-ui/components/TagOption.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import localStyles from './TagOption.css'
+
+const TagOption = ({
+    data,
+    active,
+    handleClick,
+    newTag,
+    addTagsToReverseDoc,
+}) => (
+    <div
+        className={localStyles.menuItem}
+        onClick={() =>
+            newTag === 1 ? addTagsToReverseDoc(data) : handleClick(data)}
+    >
+        {newTag === 1 ? 'add new: ' + data : data}
+        {active && <i className="material-icons">done</i>}
```

Comment:
All these conditions are essentially derived state that should be done at higher levels in the React tree. Either in the container or, more ideally as you can memoize, in selectors. The views should ideally just be pure rendering of the props given to them without having to make any decisions.

Reply:
This active prop is used to check that this option is selected or not, but I think if we'll use change the resultTags then there is no need to use have active prop.